packages/core/src/common/uri.ts

```diff
@@ -27,7 +27,7 @@ export class URI {
         return new URI(Uri.file(path));
     }
 
-    public static isUri(uri: unknown): boolean {
+    public static isUri(uri: unknown): uri is URI {
```

Comment:
Suggestion: I don't think this typing is actually correct. It doesn't try to identify whether the `uri is URI` but rather `uri is Uri` (which is weird, given in what class this method is).